File: C-Sharp/Solução de problemas em C#/3-Encaixa ou Não/solucao.cs
/* Paulinho tem em suas mãos um novo problema. Agora a sua professora lhe pediu que construísse um programa para verificar, à partir de dois valores muito grandes A e B, se B corresponde aos últimos dígitos de A.

Entrada
A entrada consiste de vários casos de teste. A primeira linha de entrada contém um inteiro N que indica a quantidade de casos de teste. Cada caso de teste consiste de dois valores A e B maiores que zero, cada um deles podendo ter até 1000 dígitos.

Saída
Para cada caso de entrada imprima uma mensagem indicando se o segundo valor encaixa no primeiro valor, confome exemplo abaixo.
 */


using System;
using System.IO;
using System.Linq;

namespace desafioOne3

{

class Program

{

static void Main(string[] args)

{

int qt = int.Parse(Console.ReadLine());

for(int i=1; i<=qt; i++)

{

string[] v = Console.ReadLine().Split(' ');

string a = v[0];

string b = v[1];


if (!a.EndsWith(b) )

{

Console.WriteLine("nao encaixa");

}

else if(a.EndsWith(b))

{

Console.WriteLine("encaixa");

}

else{

Console.WriteLine("não encaixa");

}

}

}

}

} 

File: Java/Introdução a Programação com Java/3-Encaixa ou Não I/solucao.java
/* Paulinho tem em suas mãos um pequeno problema. A professora lhe pediu que ele construísse um programa para verificar, à partir de dois valores inteiros A e B, se B corresponde aos últimos dígitos de A.

Entrada
A entrada consiste de vários casos de teste. A primeira linha de entrada contém um inteiro N que indica a quantidade de casos de teste. Cada caso de teste consiste de dois inteiros A (1 ≤ A < 231 ) e B (1 ≤ B < 231) positivos.

Saída
Para cada caso de entrada imprima uma mensagem indicando se o segundo valor encaixa no primeiro valor, confome exemplo abaixo.

 
Exemplo de Entrada	Exemplo de Saída
4											encaixa
5678690 78690					nao encaixa
5434554 543						encaixa
1243 1243							nao encaixa
54 654
 */



import java.util.Scanner;

public class App {

	public static void main(String[] args) throws Exception {
		Scanner scan = new Scanner(System.in);
		
		int count = scan.nextInt();
			for (int i = 0; i < count; i++) {
		String A = scan.next();
		String B = scan.next();
			//complete a resolução do problema com sua solução

			if (A.endsWith(B)) System.out.println("encaixa");
			else System.out.println("nao encaixa");
		}

	}
}